### PR TITLE
[ blas/OpenCL ] SGEMM OpenCL kernels added

### PR DESCRIPTION
**Pull request by s-debadri**
Added SGEMM OpenCL kernels for the following:

- `noTrans`
- `transA`
- `transB`
- `transAB`

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Debadri Samaddar <s.debadri@samsung.com>

### CODE REVIEW

File: test/unittest/unittest_blas_kernels_cl.cpp [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+  nntrainer::Tensor B_fp32(batch, channel, height_b, width_b, t_type_nchw_fp32);
+
+  GEN_TEST_INPUT(A, ((i * (batch * height * channel) + j * (batch * height) +
+                      k * (width) + l + 1) %
+                     MOD) *
+                      alpha);
+  GEN_TEST_INPUT_B(B, ((i * (batch * height_b * channel) +
+                        j * (batch * height_b) + k * (width_b) + l + 1) %
+                       MOD) *
+                        alpha);
```
Comment (EunjuYang) [inline test/unittest/unittest_blas_kernels_cl.cpp:206]: Was it intended to skip testing on fp16?

Comment (s-debadri) [reply test/unittest/unittest_blas_kernels_cl.cpp:206]: Running tests on fp32 only as of now

File: test/unittest/unittest_blas_kernels_cl.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ TEST(blas_kernels, dotCL_sgemv) {
   int width = 768;
 
   int height_b = 768;
-  int width_b = 96000;
+  int width_b = 2048;
```
Comment (djeong20) [inline test/unittest/unittest_blas_kernels_cl.cpp:47]: is there any particular reason to change this?

**Review by djeong20 — APPROVED** [review]
LGTM!

File: test/unittest/unittest_blas_kernels_cl.cpp (hunk 2 above)
Comment (s-debadri) [reply test/unittest/unittest_blas_kernels_cl.cpp:47]: No reason as such. Was testing with different values. Might have changed it to run this test faster to check other test results quickly.

**Review by EunjuYang — APPROVED** [review]

**Pull request merged**